src/default-xpring-client.ts

```diff
@@ -143,39 +159,54 @@ class DefaultXpringClient implements XpringClientDecorator {
       throw new Error(XpringClientErrorMessages.xAddressRequired)
     }
 
-    const normalizedAmount = bigInt(amount.toString())
+    const normalizedDrops = drops.toString()
 
     const fee = await this.getMinimumFee()
     const accountData = await this.getAccountData(classicAddress.address)
     const lastValidatedLedgerSequence = await this.getLastValidatedLedgerSequence()
 
     const xrpDropsAmount = new XRPDropsAmount()
-    xrpDropsAmount.setDrops(normalizedAmount.toJSNumber())
+    xrpDropsAmount.setDrops(normalizedDrops)
 
     const currencyAmount = new CurrencyAmount()
     currencyAmount.setXrpAmount(xrpDropsAmount)
 
-    const destinationAccount = new AccountAddress()
-    destinationAccount.setAddress(destination)
+    const amount = new Amount()
+    amount.setValue(currencyAmount)
 
-    const payment = new Payment()
-    payment.setDestination(destinationAccount)
-    payment.setAmount(currencyAmount)
+    const destinationAccountAddress = new AccountAddress()
+    destinationAccountAddress.setAddress(destinationAddress)
+
+    const destination = new Destination()
+    destination.setValue(destinationAccountAddress)
 
     const account = new AccountAddress()
     account.setAddress(sender.getAddress())
 
+    const account2 = new Account()
+    account2.setValue(account)
+
+    const payment = new Payment()
+    payment.setDestination(destination)
+    payment.setAmount(amount)
+
+    const lastLedgerSequence = new LastLedgerSequence()
+    lastLedgerSequence.setValue(
+      lastValidatedLedgerSequence + maxLedgerVersionOffset,
+    )
+
+    const signingPublicKeyBytes = Utils.toBytes(sender.getPublicKey())
+    const signingPublicKey = new SigningPublicKey()
+    signingPublicKey.setValue(signingPublicKeyBytes)
+
     const transaction = new Transaction()
-    transaction.setAccount(account)
+    transaction.setAccount(account2)
```

Comment:
What is `account2` here? Why did this switch from `account` to `account2`? What is `account`?

Misc: Can we name `account2` something more descriptive?

Reply:
Yeah, sorry about that. leftover code from prototyping.